src/components/InputV2/Select.tsx

```diff
@@ -0,0 +1,132 @@
+import PropTypes from 'prop-types'
+import { useEffect } from 'react'
+import { Label } from './Label'
+import truncateAddress from '../../utils/truncateAddress'
+
+export const Select = ({
+  vertical,
+  centered,
+  label,
+  info,
+  autoFocus,
+  disabled,
+  address,
+  required,
+  placeholder,
+  options,
+  value,
+  onChange,
+  onFocus,
+  onBlur,
+  onEnter,
+  setIsValid
+}: SelectProps) => {
+  // Unless there is a placeholder option, ensure value
+  // is set to the first supplied option when it's empty
+  useEffect(
+    () => !placeholder && !value && options[0] && onChange(options[0]),
+    [value, placeholder, options, onChange]
+  )
+
+  // Communicate validity to parent component
+  useEffect(
+    () => setIsValid(!required || !!value),
+    [value, required, setIsValid]
+  )
+
+  return (
+    <Label disabled={disabled} vertical={vertical} centered={centered}>
+      {vertical ? (
+        <>
+          {label && <span>{label}</span>}
+          {info && <span className='info'>{info}</span>}
+        </>
+      ) : (
+        <div>
+          {label && <span>{label}</span>}
+          {info && <span className='info'>{info}</span>}
+        </div>
+      )}
+      <div className='select-wrapper'>
+        <select
+          autoFocus={autoFocus}
+          disabled={disabled}
+          value={value}
+          onChange={e => onChange(e.target.value)}
+          onFocus={onFocus}
+          onBlur={onBlur}
+          onKeyDown={e => e.key === 'Enter' && onEnter()}
+        >
+          {placeholder && (
+            <option value='' disabled selected hidden>
+              {placeholder}
+            </option>
+          )}
+          {options.map(option => (
+            <option key={option} value={option}>
+              {address ? truncateAddress(option) : option}
+            </option>
+          ))}
+        </select>
+        <div className='select-arrow' />
+      </div>
+    </Label>
+  )
+}
+
+export interface SelectProps {
```

Comment:
Concerning the snapshots, I'm not 100% sure how the renderer works, but it doesn't seem to care about what is in the HTML for what CSS it will render. The CSS for all buttons seems to be the same, regardless of whether they contain a unit, delete button, or `select-wrapper`  in this case. I think it only renders the CSS here due to the use of styled components. Note that styles coming from `base-css` aren't rendered here either.

Do you mean extending the `BaseInputProps`?

I realize there is a bit of duplication in the props and how they are used, but extending the `BaseInputProps` for `SelectProps` feels wrong to me. Because the `Select` component does not inherit from `BaseInput`, they shouldn't necessarily affect eachother. Say that we use the `BaseInputProps`, strip what we don't need for `Select` and add what we do need, that might save a few lines of code now. But if we modify `BaseInput` later that, that means we always have to be aware of how that affects `Select`, which feels bad for unrelated components. It's different for components like `NumberInput` which actually extend `BaseInput`.